echidna/core/fill_spectrum.py

```diff
@@ -132,3 +133,95 @@ def fill_mc_spectrum(filename, spectrumname, T):
 
     return spectrum
     
+
+def fill_reco_ntuple_spectrum(filename, spectrumname, T):
+    """This function fills in the ndarray of energies, radii, times 
+    and weights. It takes the reconstructed energies and positions
+    of the events from the ntuple. In order to keep the statistics, 
+    the time dependence is performed via adding weights to every event
+    depending on the time period. Both, studied time and Half-life must 
+    be written in the same units.  
+    
+    Args:
+      filename (str): The ntuple to study 
+      spectrumname (str): The name of future ndarray 
+      T (float): The Half-life of a studied background
+
+      Returns:
+        spectrum (:class:`echidna.core.spectra.Spectra`) 
+    """
+    print filename
+    print spectrumname
+    chain=TChain("output")
+    chain.Add(filename)
+    spectrum = spectra.Spectra(str(spectrumname), chain.GetEntries())
+
+    times = [0]
+    for time_step in range(0, spectrum._time_bins):
+        time = time_step * spectrum._time_width + spectrum._time_low 
+        times.append(time)
+
```

Comment:
@jwaterfield the range of times is correct (without the extra zero - now fixed). Each event is copied into all time bins but with different weights applied.